h/indexer.py

```diff
@@ -13,27 +13,40 @@
 
 log = logging.getLogger(__name__)
 
+SETTING_NEW_INDEX = u'reindex.new_index'
+
 
 def reindex(session, es, request):
     """Reindex all annotations into a new index, and update the alias."""
 
     if get_aliased_index(es) is None:
         raise RuntimeError('cannot reindex if current index is not aliased')
 
+    settings = request.find_service(name='settings')
+
     new_index = configure_index(es)
-    indexer = BatchIndexer(session, es, request, target_index=new_index, op_type='create')
 
-    errored = indexer.index()
-    if errored:
-        log.debug('failed to index {} annotations, retrying...'.format(
-            len(errored)))
-        errored = indexer.index(errored)
-        if errored:
-            log.warn('failed to index {} annotations: {!r}'.format(
-                len(errored),
-                errored))
+    try:
+        settings.put(SETTING_NEW_INDEX, new_index)
+        request.tm.commit()
```

Comment:
Hmm. Ideally this would use `session.commit()` rather than `request.tm.commit()`, but I can see how that's not going to work right now...

...gah. Not your problem, and it'll do for now.